src/kouros/pos/grpc_manager.go

```diff
@@ -73,8 +73,209 @@ func (p *POSGRPCManager) ListDevices() (*pb.ListDeviceResponse, error) {
 //     - solution: Solution of the problem occured, if any (string)
 // info: info object contains the following fields
 //    1. version: PoseidonOS version (string)
-func (p *POSGRPCManager) CreateArray(param pb.CreateArrayRequest_Param) (*pb.CreateArrayResponse, error) {
+func (p *POSGRPCManager) CreateArray(param *pb.CreateArrayRequest_Param) (*pb.CreateArrayResponse, error) {
 	command := "CREATEARRAY"
-	req := &pb.CreateArrayRequest{Command: command, Rid: utils.GenerateUUID(), Requestor: "cli", Param: &createArrayPB}
+	req := &pb.CreateArrayRequest{Command: command, Rid: utils.GenerateUUID(), Requestor: p.requestor, Param: param}
 	return grpc.SendCreateArray(p.connection, req)
 }
+
+// Add device command add spare device in PoseidonOS Array
+// The function takes a byte array in JSON format as parameter and returns response in JSON format
```

Comment:
Please correct the description. Now the Parameters are not in JSON format

Reply:
oh Thanks for pointing out, I will correct the description